src/sentry/models/organizationslugreservation.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import annotations
+
+from enum import IntEnum
+from typing import Any, Collection, List, Mapping
+
+from django.db import models
+from django.utils import timezone
+from sentry_sdk import capture_exception
+
+from sentry.backup.scopes import RelocationScope
+from sentry.db.models.base import control_silo_only_model, sane_repr
+from sentry.db.models.fields import BoundedBigIntegerField
+from sentry.db.models.outboxes import ReplicatedControlModel
+from sentry.models.outbox import ControlOutboxBase, OutboxCategory
+from sentry.services.hybrid_cloud import REGION_NAME_LENGTH
+from sentry.utils.env import in_test_environment
+
+
+class OrganizationSlugReservationType(IntEnum):
+    PRIMARY = 0
+    VANITY_ALIAS = 1
+    TEMPORARY_RENAME_ALIAS = 2
+
+    @classmethod
+    def as_choices(cls):
+        return [(i.value, i.value) for i in cls]
+
+
+@control_silo_only_model
+class OrganizationSlugReservation(ReplicatedControlModel):
+    __relocation_scope__ = RelocationScope.Excluded
+    category = OutboxCategory.ORGANIZATION_SLUG_RESERVATION_UPDATE
+    replication_version = 1
+
+    slug = models.SlugField(unique=True, null=False)
+    organization_id = BoundedBigIntegerField(db_index=True, null=False)
+    user_id = BoundedBigIntegerField(db_index=True, null=False)
```

Comment:
```suggestion
    user_id = BoundedBigIntegerField(db_index=True, null=False)
```

Should we have this be a foreign key to the `User` table? Both `OrganizationSlugReservation` and `User` reside in the control silo.

Reply:
I don't think so. If a user is deleted and was the original creator of the slug, we don't want this constraint active.